internal/nexodus/nexodus.go

```diff
@@ -489,6 +520,66 @@ func (ax *Nexodus) Stop() {
 	}
 }
 
+// reconcileSecurityGroups will check the security group and update it if necessary.
+func (ax *Nexodus) reconcileSecurityGroups(ctx context.Context) {
+	if runtime.GOOS != Linux.String() {
+		return
+	}
+
+	// lookup this device in the device cache and check if the security-group ID is nil
+	existing, ok := ax.deviceCache[ax.wireguardPubKey]
+	if ok {
+		if existing.device.SecurityGroupIds == uuid.Nil.String() {
+			ax.securityGroup = nil
+			if err := ax.processSecurityGroupRules(); err != nil {
+				ax.logger.Error(err)
+			}
+			return
+		}
+	}
+
+	// TODO: This needs some scrutiny, ax.securityGroup could be nil, empty with a uuid.nil 0000000-000.. or a valid UUID.
+	// are there any scenarios where a group has been created/updated/patched that the device is not
+	// updated about via reconcile while still maintaining nil checks?
+	if ax.securityGroup == nil {
+		return
+	}
+
+	// if the security group ID is not nil, lookup the ID and check for any changes
+	responseSecGroup, _, err := ax.client.SecurityGroupApi.GetSecurityGroup(ctx, ax.orgId, ax.securityGroup.Id).Execute()
+	if err != nil {
+		// if the group ID returns a 404, clear the current rules
+		if strings.Contains(err.Error(), securityGroupNotFound.Error()) {
+			if err := ax.processSecurityGroupRules(); err != nil {
+				ax.logger.Error(err)
+			}
+			ax.securityGroup = nil
+			return
+		} else {
+			ax.logger.Debugf("Error retrieving the security group: %v", err)
+		}
+	}
+
+	if responseSecGroup == nil {
+		ax.logger.Debug("Security group is not set.")
+		return
+	}
+
+	if !reflect.DeepEqual(responseSecGroup, ax.securityGroup) {
+		ax.logger.Debugf("Security Group change detected: %+v", responseSecGroup)
+		if responseSecGroup.Id != ax.securityGroup.Id ||
+			!reflect.DeepEqual(responseSecGroup.InboundRules, ax.securityGroup.InboundRules) ||
+			!reflect.DeepEqual(responseSecGroup.OutboundRules, ax.securityGroup.OutboundRules) {
```

Comment:
For sure, they are triggered. I added a test since its a good one to validate that just changes an octet of a prefix.

Here is an example of a rule update and the corresponding rule changes for both port and prefix seperately:

```
./nexctl \
    --host https://api.try.nexodus.127.0.0.1.nip.io --username admin --password floofykittens security-group update \
    --name="default" --description="security group testing" \
    --inbound-rules='[{"ip_protocol": "tcp", "from_port": 22, "to_port": 22, "ip_ranges": ["100.100.0.0/16"]}]' \
    --outbound-rules='[{"ip_protocol": "tcp"}]' \
   --security-group-id="${SECURITY_GROUP_ID}" \
   --organization-id="${ORGANIZATION_ID}"

brent@ub-arm:~/stun$ sudo nft list ruleset
table inet nexodus {
	chain nexodus-inbound {
		type filter hook input priority filter; policy accept;
		ct state established,related iifname "wg0" counter packets 0 bytes 0 accept
	}

	chain nexodus-outbound {
		type filter hook input priority filter; policy accept;
	}
}

./nexctl \
    --host https://api.try.nexodus.127.0.0.1.nip.io --username admin --password floofykittens security-group update \
    --name="default" --description="security group testing" \
    --inbound-rules='[{"ip_protocol": "tcp", "from_port": 22, "to_port": 22, "ip_ranges": ["100.101.0.0/16"]}]' \
    --outbound-rules='[{"ip_protocol": "tcp"}]' \
   --security-group-id="${SECURITY_GROUP_ID}" \
   --organization-id="${ORGANIZATION_ID}"


brent@ub-arm:~/stun$ sudo nft list ruleset
table inet nexodus {
	chain nexodus-inbound {
		type filter hook input priority filter; policy accept;
		ct state established,related iifname "wg0" counter packets 0 bytes 0 accept
		ip saddr 100.100.0.0/16 tcp dport 22 iifname "wg0" counter packets 0 bytes 0 accept
		iifname "wg0" counter packets 0 bytes 0 drop
	}

	chain nexodus-outbound {
		type filter hook input priority filter; policy accept;
		meta nfproto ipv4 tcp dport 0-65535 iifname "wg0" counter packets 0 bytes 0 accept
		meta nfproto ipv6 tcp dport 0-65535 iifname "wg0" counter packets 0 bytes 0 accept
		iifname "wg0" counter packets 0 bytes 0 drop
	}
}


./nexctl \
    --host https://api.try.nexodus.127.0.0.1.nip.io --username admin --password floofykittens security-group update \
    --name="default" --description="security group testing" \
    --inbound-rules='[{"ip_protocol": "tcp", "from_port": 22, "to_port": 23, "ip_ranges": ["100.101.0.0/16"]}]' \
    --outbound-rules='[{"ip_protocol": "tcp"}]' \
   --security-group-id="${SECURITY_GROUP_ID}" \
   --organization-id="${ORGANIZATION_ID}"


brent@ub-arm:~/stun$ sudo nft list ruleset
table inet nexodus {
	chain nexodus-inbound {
		type filter hook input priority filter; policy accept;
		ct state established,related iifname "wg0" counter packets 0 bytes 0 accept
		ip saddr 100.101.0.0/16 tcp dport 22-23 iifname "wg0" counter packets 0 bytes 0 accept
		iifname "wg0" counter packets 0 bytes 0 drop
	}

	chain nexodus-outbound {
		type filter hook input priority filter; policy accept;
		meta nfproto ipv4 tcp dport 0-65535 iifname "wg0" counter packets 0 bytes 0 accept
		meta nfproto ipv6 tcp dport 0-65535 iifname "wg0" counter packets 0 bytes 0 accept
		iifname "wg0" counter packets 0 bytes 0 drop
	}
}
```